lib/ldp.js

```diff
@@ -256,25 +256,53 @@ class LDP {
     })
   }
 
-  put (host, resourcePath, stream, callback) {
+  isValidRdf (body, requestUri, contentType) {
+    const resourceGraph = $rdf.graph()
+    try {
+      $rdf.parse(body, resourceGraph, requestUri, contentType)
+    } catch (err) {
+      debug.ldp('VALIDATE -- Error parsing data: ' + err)
+      return false
+    }
+    return true
+  }
+
+  _put (host, resourcePath, callback, resourceHandler) {
     const ldp = this
     const root = !ldp.multiuser ? ldp.root : ldp.root + host + '/'
     const filePath = utils.uriToFilename(resourcePath, root, host)
 
     // PUT requests not supported on containers. Use POST instead
     if (filePath.endsWith('/')) {
-      return callback(error(409,
-        'PUT not supported on containers, use POST instead'))
+      return callback(error(409, 'PUT not supported on containers, use POST instead'))
     }
+
     // First, create the enclosing directory, if necessary
     const dirName = path.dirname(filePath)
     mkdirp(dirName, (err) => {
       if (err) {
         debug.handlers('PUT -- Error creating directory: ' + err)
-        return callback(error(err,
-          'Failed to create the path to the new resource'))
+        return callback(error(err, 'Failed to create the path to the new resource'))
       }
+
       // Directory created, now write the file
+      resourceHandler(filePath)
+    })
+  }
+
+  putText (host, resourcePath, text, callback) {
+    this._put(host, resourcePath, callback, (filePath) => {
+      try {
+        fs.writeFileSync(filePath, text)
```

Comment:
Async?

Reply:
I'm wondering about the difference  of `fs.writeFileSync` and `await fs.writeFile` in general. It seems that there's [no difference in blocking IO](https://www.daveeddy.com/2013/03/26/synchronous-file-io-in-nodejs/), and it seems that it doesn't affect [the main thread as filesystem operations are handled in worker threads](http://disq.us/p/16tx816).

It doesn't seem to be that much of a difference, so in this case I think I prefer `fs.writeFileSync`. But at some point we might want to have a convention on this, so that we stay consistent throughout the code.